### PR TITLE
Added configuration & env. setup to readme.

### LINKED ISSUE #224: Installation Guide
A noob-friendly guide for configuring a fresh installation including the database migration run procedure would be useful.

### PR DESCRIPTION
**Pull request by pavlos-io**
Fixes #224.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,13 +20,11 @@ End of Codes was developed by:
...
-====
-You can read more about the development of the game on our [blog](http://blog.endofcodes.com)
-
-Technology
+Configuration
 ==========
-End of Codes is written in HTML, CSS, Javascript, PHP, and MySQL. We require:
+End of Codes is written in HTML, CSS, Javascript, PHP, and MySQL.
+
+### Requires
```
Comment (dionyziz) [inline README.md]: "Requirements"

File: README.md [hunk 2]
```diff
@@ -20,13 +20,11 @@ End of Codes was developed by:
 
 If you're interested in contributing, just fork, fix a bug or build a feature, and pull request.
 
-Blog
-====
-You can read more about the development of the game on our [blog](http://blog.endofcodes.com)
-
-Technology
+Configuration
```
Comment (dionyziz) [inline README.md]: "Installation"

File: README.md [hunk 3]
```diff
@@ -20,13 +20,11 @@ End of Codes was developed by:
...
 ==========
-End of Codes is written in HTML, CSS, Javascript, PHP, and MySQL. We require:
+End of Codes is written in HTML, CSS, Javascript, PHP, and MySQL.
```
Comment (dionyziz) [inline README.md]: This line is somewhat unrelated to installation / configuration; also it is more relevant to what the project is about, so it should be moved to the end of the about section.

File: README.md [hunk 4]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
 
 End of Codes has been tested under MySQL 5.5 and 5.6 and Apache 2.2, but it may work with other versions also.
 
+### Environment Setup
+
+- Create a new database called 'endofcodes'. 
+- Import the **database/db_schema.sql** into the db through phpMyAdmin, or dump it through the command line. Like so: `mysql -u user -p pass endofcodes < absolute/path/to/dump/db_schema.sql`
```
Comment (dionyziz) [inline README.md]: The canonical way of installing is through migrations, not through db_schema. Perhaps db_schema should be dropped altogether. What's your take on this, @VitSalis @pkakelas ?

File: README.md [hunk 5]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
 
 End of Codes has been tested under MySQL 5.5 and 5.6 and Apache 2.2, but it may work with other versions also.
 
+### Environment Setup
+
+- Create a new database called 'endofcodes'. 
+- Import the **database/db_schema.sql** into the db through phpMyAdmin, or dump it through the command line. Like so: `mysql -u user -p pass endofcodes < absolute/path/to/dump/db_schema.sql` 
+
+- Open **config/config.php** and set your 'user' and 'pass' for an existing user in the database.
```
Comment (dionyziz) [inline README.md]: "For an existing user" is quite vague. The user must have full permissions to the database, especially if they're going to run migrations. Suggest that we instruct the creation of a new user with permissions only in the database in question.

File: README.md [hunk 6]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+    'db' => [
+        'host' => 'localhost',
+        'user' => 'endofcodes',
+        'pass' => 'sample_pass',
+        'dbname' => 'endofcodes'
+    ],
+```
+
+### Vhost Setup
+- Vhost is a custom domain name on your local server, that points to a directory. This way you can identify and access your local projects through reasonable naming in the browser.
```
Comment (dionyziz) [inline README.md]: Unnecessary "," after "local server".

File: README.md (hunk 3 above)
Comment (gtklocker) [inline README.md]: No need to mention HTML/CSS/JS in any way since they're all front-end.

File: README.md [hunk 7]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+- If you're on Mac, w/ MAMP & Ports set to MySQL defaults, open **/private/etc/hosts** and add `127.0.0.1  endofgames.loc`. Like so:
```
Comment (dionyziz) [inline README.md]: Please spell out "with" fully.

Comment (dionyziz) [inline README.md]: The fact that the instructions are for mac should not be in small print. Please make separate sections for instructions per operating system.

File: README.md [hunk 8]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
 
 End of Codes has been tested under MySQL 5.5 and 5.6 and Apache 2.2, but it may work with other versions also.
 
+### Environment Setup
+
+- Create a new database called 'endofcodes'.
```
Comment (dionyziz) [inline README.md]: Use `*` instead of `-` in markdown for unordered lists. `-` is used by git diffs to indicate that a line was removed, so it's quite confusing.

File: README.md [hunk 9]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+    ],
+```
+
+### Vhost Setup
+- Vhost is a custom domain name on your local server, that points to a directory. This way you can identify and access your local projects through reasonable naming in the browser.
+- If you're on Mac, w/ MAMP & Ports set to MySQL defaults, open **/private/etc/hosts** and add `127.0.0.1  endofgames.loc`. Like so:
+
+```
+127.0.0.1       localhost
+127.0.0.1       endofgames.loc
```
Comment (dionyziz) [inline README.md:63]: This is true only if the configuration is for a local service and not made available publicly. However, these instructions should be more general.

Comment (gtklocker) [inline README.md:63]: This is going to be confusing. Just append a hosts line for endofcodes in the end. Maybe you can also assume that the reader knows how to work with vhosts.

File: README.md [hunk 10]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+```
+Open **Applications/MAMP/conf/apache/httpd.conf** and uncomment the line `#Include /Applications/MAMP/conf/apache/extra/httpd-vhosts.conf`
+
+Next open **Applications/MAMP/conf/apache/extra/httpd-vhosts.conf** and add this: 
+```
+<VirtualHost *:80>
+ServerName endofcodes.loc
```
Comment (gtklocker) [inline README.md]: Indentation.

File: README.md [hunk 11]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+DocumentRoot absolute/path/to/folder/endofcodes
+<Directory absolute/path/to/folder/endofcodes>
+Options Indexes FollowSymLinks
```
Comment (dionyziz) [inline README.md]: Not sure why symlink following is needed?

File: README.md [hunk 12]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+AllowOverride All
+Order allow,deny
+Allow from all
+#Require all granted
```
Comment (dionyziz) [inline README.md]: What is this?

File: README.md [hunk 13]
```diff
@@ -37,6 +35,58 @@ for arrays. PHP 5 is required because we use OOP5 features.
 
 End of Codes has been tested under MySQL 5.5 and 5.6 and Apache 2.2, but it may work with other versions also.
 
+### Environment Setup
+
```
Comment (dionyziz) [inline README.md:40]: A first step to fork and clone may be appropriate. Perhaps a reference to CONTRIBUTING.md and a mention that a folder will be created at this point.

File: README.md (hunk 4 above)
Comment (vitsalis) [inline README.md]: I think we should keep the db_schema file so that users have an alternative option.

File: README.md [hunk 14]
```diff
@@ -37,6 +36,59 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+* Fork the repository & Clone it on your machine. This will automatically create a new folder (in the current working directory) containing the repo.
```
Comment (dionyziz) [inline README.md:41]: Let's include a link to CONTRIBUTING.md which has more extensive instructions regarding this step. Also "clone" should be lower-case, as it's not at the beginning of the sentence (and there's no reason for capitalization, as the command is also lower-case).

File: README.md [hunk 15]
```diff
@@ -37,6 +36,59 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+    'db' => [
+        'host' => 'localhost',
+        'user' => 'endofcodes',
+        'pass' => 'sample_pass',
+        'dbname' => 'endofcodes'
+    ],
+```
+
+### Vhost Setup
+* Vhost is a custom domain name on your local server, that points to a directory. This way you can identify and access your local projects through reasonable naming in the browser.
```
Comment (dionyziz) [inline README.md:58]: Unnecessary `,`

File: README.md [hunk 16]
```diff
@@ -37,6 +36,59 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+* If you're on Mac, w/ MAMP & Ports set to MySQL defaults, open **/private/etc/hosts** and add `127.0.0.1  endofgames.loc`. Like so:
```
Comment (dionyziz) [inline README.md:59]: Spell out "w/" as "with".

File: README.md [hunk 17]
```diff
@@ -37,6 +36,59 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+'development' => [
+    'db' => [
+        'host' => 'localhost',
+        'user' => 'endofcodes',
+        'pass' => 'sample_pass',
+        'dbname' => 'endofcodes'
+    ],
+```
+
+### Vhost Setup
```
Comment (dionyziz) [inline README.md:57]: Create a separate section for mac and have a clean headline that says `Mac`.

File: README.md [hunk 18]
```diff
@@ -37,6 +36,59 @@ for arrays. PHP 5 is required because we use OOP5 features.
...
+    Allow from all
+    #Require all granted
+    </Directory>
+    ErrorLog endofcodes.log
+    CustomLog endofcodes.log combined
+</VirtualHost> 
+```
+Restart MAMP and open http://endofcodes.loc in the broswer.
+
+* If you're on Linux [this](https://www.digitalocean.com/community/articles/how-to-set-up-apache-virtual-hosts-on-ubuntu-12-04-lts) tutorial explains a similar process.
```
Comment (dionyziz) [inline README.md:85]: Create a separate `Linux` subsection with just this line.

**Comment by pkakelas** [comment]
I believe that this pull request must change in some parts in case that #293 gets merged first.

**Comment by dionyziz** [comment]
As #293 is now merged, these instructions are now outdated. Recommend updating them or opening a new pull request.

File: README.md [hunk 19]
```diff
@@ -37,6 +36,59 @@ for arrays. PHP 5 is required because we use OOP5 features.
 
 End of Codes has been tested under MySQL 5.5 and 5.6 and Apache 2.2, but it may work with other versions also.
 
+### Environment Setup
+
+* Fork the repository & Clone it on your machine. This will automatically create a new folder (in the current working directory) containing the repo.
+* Create a new database called 'endofcodes'. 
+* Import the **database/db_schema.sql** into the db through phpMyAdmin, or dump it through the command line. Like so: `mysql -u user -p pass endofcodes < absolute/path/to/dump/db_schema.sql` 
+
+* Open **config/config.php** and set your 'user' and 'pass' for an existing user in the database.
```
Comment (bugos) [inline README.md:45]: Open `config/config-local.php`

Comment (bugos) [inline README.md:45]: Alternatively you can open endofcodes in a web browser (`localhost/endofcodes/`) and fill in the database details.

**Comment by dionyziz** [comment]
@bugos Can you take ownership of this PR?

**Comment by pkakelas** [comment]
@ProjectPxl superseded by this.

**Comment by dionyziz** [comment]
Superseded by #363
